internal/server/data/destination.go

```diff
@@ -38,51 +75,42 @@ func ListDestinations(db GormTxn, p *Pagination, selectors ...SelectorFunc) ([]m
 	return list[models.Destination](db, p, selectors...)
 }
 
-func DeleteDestinations(db GormTxn, selector SelectorFunc) error {
-	toDelete, err := ListDestinations(db, nil, selector)
-	if err != nil {
-		return err
-	}
-
-	if len(toDelete) > 0 {
-		ids := make([]uid.ID, 0)
-		for _, g := range toDelete {
-			ids = append(ids, g.ID)
-		}
-
-		return deleteAll[models.Destination](db, ByIDs(ids))
-	}
-
-	return internal.ErrNotFound
```

Comment:
Previously we were returning `ErrNotFound` if the ID was not found, but we don't do that for any other types. I suspect we should be consistent with the behaviour. 

If we want to return a not found on a missing delete then we can check `result.RowsAffected() > 0` from the result returned by `tx.Exec`.